identity/roles.go

```diff
@@ -189,3 +190,117 @@ func (c *RolesClient) Delete(ctx context.Context, input *DeleteRoleInput) error
 
 	return nil
 }
+
+//
+type RoleTags struct {
+	//
+	Name string `json:"name"`
+	//
+	RoleTag []string `json:"role-tag"`
+}
+
+//
+type GetRoleTagsInput struct {
+	//
+	ResourceType string
+	//
+	ResourceID string
+}
+
+//
+func (c *RolesClient) GetRoleTags(ctx context.Context, input *GetRoleTagsInput) (*RoleTags, error) {
+	fullPath := path.Join("/", c.client.AccountName, input.ResourceType, input.ResourceID)
+	reqInputs := client.RequestInput{
+		Method: http.MethodGet,
+		Path:   fullPath,
+	}
+
+	if err := validateResourceType(input.ResourceType); err != nil {
+		return nil, err
+	}
+
+	response, err := c.client.ExecuteRequestRaw(ctx, reqInputs)
+	if response == nil {
+		return nil, errors.Wrap(err, "unable to get role tags")
+	}
+	if response.Body != nil {
+		defer response.Body.Close()
+	}
+
+	result := &RoleTags{
+		Name: reqInputs.Path,
+	}
+
+	//
+	if roleTag := response.Header.Get("Role-Tag"); roleTag != "" {
+		result.RoleTag = strings.Split(roleTag, ",")
+	}
+
+	return result, nil
+}
+
+//
+type SetRoleTagsInput struct {
+	//
+	ResourceType string
+	//
+	ResourceID string
+	//
+	RoleTag []string `json:"role-tag"`
+}
+
+//
+func (c *RolesClient) SetRoleTags(ctx context.Context, input *SetRoleTagsInput) (*RoleTags, error) {
+	fullPath := path.Join("/", c.client.AccountName, input.ResourceType, input.ResourceID)
+	reqInputs := client.RequestInput{
+		Method: http.MethodPut,
+		Path:   fullPath,
+		Body:   input,
+	}
+
+	if err := validateResourceType(input.ResourceType); err != nil {
+		return nil, err
+	}
+
+	respReader, err := c.client.ExecuteRequest(ctx, reqInputs)
+	if respReader != nil {
+		defer respReader.Close()
+	}
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to set role tags")
+	}
+
+	var result *RoleTags
+	decoder := json.NewDecoder(respReader)
+	if err = decoder.Decode(&result); err != nil {
+		return nil, errors.Wrap(err, "unable to decode set role tags response")
+	}
+
+	return result, nil
+}
+
+//
+func validateResourceType(resourceType string) error {
```

Comment:
Would suffice for now, but needs to be changed after https://github.com/joyent/triton-go/issues/11 is closed.

Reply:
I decided to remove this and let the CloudAPI bubble an error should the resource type be different to the one supported. This removes the need to maintain any list of currently supported resources in the client.